src/main/java/net/mcreator/element/types/interfaces/IPOIProvider.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.types.interfaces;
+
+import net.mcreator.element.parts.MItemBlock;
+
+import java.util.List;
+
+public interface IPOIProvider {
+
+	/**
+	 * This method determines what POI blocks as MItemBlock list are provided by this generatable element
+	 *
+	 * @return A list of MItemBlock that are POIs provided by this generatable element
+	 */
+	List<MItemBlock> poiBlocks();

```

Comment:
I wouldn't really create a dedicated interface for such thing (similarly to `PointOfInterestValidator`) but instead incorporate this in more common `IMCItemProvider` (like `getBlocksOnly()` or simply `getBlocks()`). Is it possible to do in scope of this PR or it should be another one? (`getItemsOnly()` counterpart may be needed for consistency)

Reply:
As far as I get it, `poiBlocks()` simply returns list of blocks provided by the mod element (OKay, another type of them). If that's it, why not use the existing interface? Or should mapped and internal variants be in separate interfaces?

Reply:
From an OO perspective, it is also better to have more interfaces if they describe different functionality (which providing list of all blocks or special case where object can provide POI blocks is a different functionality).

Also from the perspective of time-investment for a more or less identical solution at the end of the day

Reply:
> Or should mapped and internal variants be in separate interfaces?

they describe different functionality (which providing list of all blocks or special case where object can provide POI blocks is a different functionality).

Reply:
>As far as I get it, `poiBlocks()` simply returns list of blocks provided by the mod element (OKay, another type of them).

Am I right with this part? Because I'd at least call this interface more generic and add the same method for items (maybe outside this PR).

Reply:
Yes, but it returns POI blocks. Reason why this is better as separate interface is that when iterating, a simple instanceof check splits blocks that can provide pois and those that can't.

If this is part of common items/blocks providers, for every block providing them, addall with empty list is executed because the interface is more generic in describing what that interface provides

Reply:
That I get (I hope), I mean, maybe rename e.g. `IPOIProvider` to `IBlockItemMappingsProvider` and `poiBlocks()` to `blockMappings()` and then also add `itemMappings()`?

Reply:
Not sure you do.

Making the interface more general would defeat what I just said. Not sure how to explain it in a different way, but I don't have time to write code example at the moment

Reply:
So not all block are provided as POI blocks?

Reply:
So not all blocks are provided as POI blocks?

Reply:
No, only "dimension portal block" and "villager workstation" blocks are POI blocks, other blocks are blocks that don't have POI.

This method is to get a list of blocks that define POI (as these blocks can't be used). I guess naming could be changed then if this was not clear enough

Reply:
~~I asked for review more to check if the functionality of POI validation works as designed, not to rethink "design concepts of MCreator" if I exaggerate a bit in this already big and delayed PR~~

Reply:
>So not all blocks are provided as POI blocks?

In this case all should be good from what I understood in the refactoring.

Reply:
Do you think there should be a better name for the interface or the method, given it seemed confusing to you? Or is Javadoc not clear enough?

Reply:
> No, only "dimension portal block" and "villager workstation" blocks are POI blocks, other blocks are blocks that don't have POI.

You forgot bee related blocks

Reply:
There are two sources:

- vanilla blocks (https://github.com/MCreator/MCreator/pull/2630/files#diff-96275434d82d294f1c4e89e2c05a9c67e38490ec3cd5c6d53600d8436843f8eb, where BEEHIVE and BEE_NEST are marked as POI)
- MCreator MEs using POIs, meaning dimensions (dimension portal block) and villager stations

So I think they (bee related blocks) are covered

Reply:
I just tried using some vanilla profession blocks as a custom POI and the system seems to work as designed.